openupgradelib/openupgrade.py

```diff
@@ -2486,6 +2487,30 @@ def update_module_moved_fields(
     )
 
 
+def safe_unlink(records, do_raise=False):
+    """Allow for errors to occur during unlinking of records.
+
+    Prevent broken database transactions, and by default, catch exceptions.
+
+    :param records: an iterable (not necessarily recordset) of records to
+    unlink.
+    :param do_raise: when set to True, don't catch the exception but only log
+    an error.
+    """
+    for record in records:
+        logger.debug("Deleting record %s#%s", record._name, record.id)
+        if not record.exists():
+            continue
+        try:
+            with record.env.cr.savepoint():
+                record.unlink()
+        except Exception as e:
+            if do_raise:
+                raise
+            logger.error("Error deleting %s#%s: %s",
```

Comment:
I think it's better to not log ERROR, as it can be something expected (and that's why you build this method). I would go just for INFO.

Reply:
OK, updated.